src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -294,9 +294,16 @@ public IActionResult GetAdminToken()
         [Route("admin/warmup")]
         [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
         [RequiresRunningHost]
-        public async Task<IActionResult> Warmup([FromServices] IScriptJobHost scriptHost)
+        public async Task<IActionResult> Warmup([FromServices] WebJobsScriptHostService hostService)
         {
-            await scriptHost.TryInvokeWarmupAsync();
+            var jobHost = hostService.Services?.GetService<IScriptJobHost>();
+            if (jobHost == null)
+            {
+                _logger.LogError($"Could not find a registered {nameof(IScriptJobHost)}");
```

Comment:
This is too low level an error message for public http consumers I think. How about "No active host available."?

Reply:
Right, that's a good call.

Reply:
Perhaps the status code in this case should be 503, which is what we return for other temporary unavailability cases. And return the same status code from the other 2 places where this pattern occurs.